memstore/src/test.rs

```diff
@@ -273,7 +273,13 @@ where
             voted_for: None,
         };
 
-        let initial = store.get_initial_state().await?;
+        let initial = if let Some(state) = store.get_initial_state().await?{
+            state
+        }else {
+            let state = InitialState::new_initial(NODE_ID);
+            store.save_hard_state(&state.hard_state).await?;
+            state
+        };
```

Comment:
This is just a test for `store.get_initial_state()`.

Just to check the returned value is enough.

Building a new value with `InitialState::new_initial(NODE_ID)` is only required when booting up a RaftCore.

Reply:
Gotcha